### PR TITLE
Update models.py to fix the issue #1814

### PR DESCRIPTION
**Pull request by loulo1**
This PR is to fix the issue   #https://github.com/facebook/prophet/issues/1814

I did as PyStanBackend. And now when we use the method fit of Prophet, we can do like in the documentation:

https://facebook.github.io/prophet/docs/additional_topics.html#updating-fitted-models

`
# Python
def stan_init(m):
    """Retrieve parameters from a trained model.
    
    Retrieve parameters from a trained model in the format
    used to initialize a new Stan model.
    
    Parameters
    ----------
    m: A trained model of the Prophet class.
    
    Returns
    -------
    A Dictionary containing retrieved parameters of m.
    
    """
    res = {}
    for pname in ['k', 'm', 'sigma_obs']:
        res[pname] = m.params[pname][0][0]
    for pname in ['delta', 'beta']:
        res[pname] = m.params[pname][0]
    return res

df = pd.read_csv('../examples/example_wp_log_peyton_manning.csv')
df1 = df.loc[df['ds'] < '2016-01-19', :]  # All data except the last day
m1 = Prophet().fit(df1) # A model fit to all data except the last day

%timeit m2 = Prophet().fit(df)  # Adding the last day, fitting from scratch
%timeit m2 = Prophet().fit(df, init=stan_init(m1))  # Adding the last day, warm-starting from m1
`

### CODE REVIEW

File: python/fbprophet/models.py [hunk 1]
```diff
@@ -117,17 +122,28 @@ def fit(self, stan_init, stan_data, **kwargs):
...
             kwargs['iter_warmup'] = iter_half
+        
+        args.update(kwargs)
 
-        self.stan_fit = self.model.sample(data=stan_data,
-                                          inits=stan_init,
-                                          iter_sampling=iter_half,
-                                          **kwargs)
+        args['inits'] = args['init']
+        del args['init']
```
Comment (bletham) [inline python/fbprophet/models.py:144]: same as above

**Review by bletham — COMMENTED** [review]
Thanks for the improvement! I have one minor comment about the handling of `inits` vs `init` that I think should be changed and then this is good to merge.

**Comment by loulo1** [comment]
I have just done the modifications

File: python/fbprophet/models.py [hunk 2]
```diff
@@ -117,17 +120,26 @@ def fit(self, stan_init, stan_data, **kwargs):
...
             kwargs['iter_warmup'] = iter_half
+        
+        args.update(kwargs)
 
-        self.stan_fit = self.model.sample(data=stan_data,
-                                          inits=stan_init,
-                                          iter_sampling=iter_half,
-                                          **kwargs)
+        self.stan_fit = self.model.sample(**args)
         res = self.stan_fit.sample
```
Comment (bletham) [inline python/fbprophet/models.py:143]: This needs to be changed to
```Python
res = self.stan_fit.draws()
```
in order to merge with #1849.

**Comment by bletham** [comment]
This is perfect and ready to go in, thanks for the great contribution!

There is a merge conflict that has to be resolved before I can merge it. I identified it in a comment above. I tried to make the change myself but it looks like I don't have permission to make edits to this PR. If you'd like me to make the change, there is a box on the side of this PR that says "Allow edits from maintainers", and if you check that then I'll be able to push to this PR and can fix the merge conflict (https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork). Otherwise you can also just make the change that I noted in the comment above and then it should merge cleanly.

Thanks again!

**Comment by loulo1** [comment]
Hello,

Thanks for your feedback. The box is already checked to allow you to modify the code.
image

You don't manage to modify the code?

**Comment by bletham** [comment]
Oh that's weird, it rejected it before, I must have messed something up. Thanks! I'll get this merged.

**Comment by loulo1** [comment]
Thank you :)

**Comment by jelc53** [comment]
To clarify if ``init`` then not accepted in kwargs as detailed in the docs? https://facebook.github.io/prophet/docs/additional_topics.html#updating-fitted-models

**Pull request merged**